### PR TITLE
69346 - Fix NPE when we reconcile move lines

### PR DESCRIPTION
**Pull request by paa-axelor**

### CODE REVIEW

**Review by jmi-axelor — CHANGES_REQUESTED** [review]

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineTaxServiceImpl.java [hunk 1]
```diff
@@ -299,30 +299,69 @@ public int getVatSystem(Move move, MoveLine moveline) throws AxelorException {
...
+      return null;
+    }
+
+    for (MoveLine moveLine : moveLineList) {
+      if (moveLine.getMove() != null
+          && this.isMoveLineTaxAccount(moveLine)
+          && Lists.newArrayList(
+                  MoveRepository.FUNCTIONAL_ORIGIN_PURCHASE, MoveRepository.FUNCTIONAL_ORIGIN_SALE)
+              .contains(moveLine.getMove().getFunctionalOriginSelect())
+          && moveLine.getTaxLine() == null) {
```
Comment (aol-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineTaxServiceImpl.java:361]: I think that part of the condition can be factorized with the one used in `MoveLineAttrsServiceImpl.addTaxLineRequired`

File: axelor-account/src/main/resources/views/MoveLine.xml [hunk 2]
```diff
@@ -514,15 +516,12 @@
...
       <panel name="othersPanel" title="Others"
         showIf="move.statusSelect == 2 || move.statusSelect == 3 || taxLine != null" readonly="true">
-        <field name="taxLine" canEdit="false"
-          showIf="account.accountType.technicalTypeSelect == 'income' || account.accountType.technicalTypeSelect == 'charge' || account.accountType.technicalTypeSelect == 'tax' || account.accountType.technicalTypeSelect == 'immobilisation'"
-          colSpan="6" form-view="tax-line-form" grid-view="tax-line-grid"
-          domain="self.endDate > :date OR self.endDate IS NULL"
-          readonlyIf="!account.isTaxAuthorizedOnMoveLine"
+        <field name="taxLine" canEdit="false" colSpan="6" form-view="tax-line-form"
+          grid-view="tax-line-grid" domain="self.endDate > :date OR self.endDate IS NULL"
+          readonlyIf="account.accountType.technicalTypeSelect == 'income' || account.accountType.technicalTypeSelect == 'charge' || account.accountType.technicalTypeSelect == 'tax' || account.accountType.technicalTypeSelect == 'immobilisation' || !account.isTaxAuthorizedOnMoveLine"
```
Comment (aol-axelor) [inline axelor-account/src/main/resources/views/MoveLine.xml:521]: Please use an array here to check technicalTypeSelect

File: axelor-account/src/main/resources/views/MoveLine.xml [hunk 3]
```diff
@@ -514,15 +516,12 @@
...
           onChange="account-move-line-update-tax-equiv"/>
         <field name="vatSystemSelect"
-          readonlyIf="move.statusSelect != 1 &amp;&amp; move.statusSelect != 5"
-          showIf="account.accountType.technicalTypeSelect == 'tax'"/>
+          readonlyIf="account.accountType.technicalTypeSelect != 'tax' || (move.statusSelect != 1 &amp;&amp; move.statusSelect != 5)"/>
```
Comment (aol-axelor) [inline axelor-account/src/main/resources/views/MoveLine.xml:524]: Please use an array here to check statusSelect

File: axelor-account/src/main/resources/views/MoveLine.xml [hunk 4]
```diff
@@ -762,14 +761,12 @@
...
         <field name="taxLine" canEdit="false" readonlyIf="!account.isTaxAuthorizedOnMoveLine"
-          showIf="account.isTaxRequiredOnMoveLine" requiredIf="account.isTaxRequiredOnMoveLine"
           colSpan="6" form-view="tax-line-form" grid-view="tax-line-grid"
           domain="self.endDate > :date OR self.endDate IS NULL"
           onChange="account-move-line-update-tax-equiv"/>
         <field name="account.accountType.technicalTypeSelect" hidden="true"/>
         <field name="vatSystemSelect"
-          readonlyIf="move.statusSelect != 1 &amp;&amp; move.statusSelect != 5"
-          showIf="account.accountType.technicalTypeSelect == 'tax'"
+          readonlyIf="(move.statusSelect != 1 &amp;&amp; move.statusSelect != 5) || account.accountType.technicalTypeSelect != 'tax'"
```
Comment (aol-axelor) [inline axelor-account/src/main/resources/views/MoveLine.xml:769]: Same as before

**Pull request merged**